packages/create-strapi-starter/utils/build-starter.js

```diff
@@ -0,0 +1,161 @@
+'use strict';
+
+const { resolve, join } = require('path');
+const os = require('os');
+const fse = require('fs-extra');
+const fetch = require('node-fetch');
+const chalk = require('chalk');
+const tar = require('tar');
+
+const generateNewApp = require('strapi-generate-new');
+const GitUrlParse = require('git-url-parse');
+const ora = require('ora');
+
+const {
+  runInstall,
+  runApp,
+  setupStarter,
+  initGit,
+  createInitialGitCommit,
+} = require('./child-process');
+
+function getRepoInfo(starterUrl) {
+  const repoInfo = GitUrlParse(starterUrl);
+  const { name, full_name, ref } = repoInfo;
+
+  return {
+    name,
+    full_name,
+    ref,
+  };
+}
+
+/**
+ * @param  {string} starterUrl Github url for strapi starter
+ * @param  {string} tmpDir Path to temporary directory
+ */
+async function downloadGithubRepo(starterUrl, tmpDir) {
+  const { name, full_name, ref } = getRepoInfo(starterUrl);
+
+  // TODO: Consider case for 'master'
+  const branch = ref ? ref : 'main';
```

Comment:
We should fine a way to use the default branch if one isn't specified. as the repo must be public I think it is pretty easy to get the info from the github api

Reply:
Example: https://github.com/Knutakir/default-branch

Reply:
yup a simpel fetch will do no need to add a dependency in the cli